File: sample-conference-java/app/src/main/java/com/quickblox/sample/videochat/conference/java/managers/BackgroundListener.java
package com.quickblox.sample.videochat.conference.java.managers;

import android.util.Log;

import com.quickblox.chat.QBChatService;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;

public class BackgroundListener implements LifecycleObserver {
    private static final String TAG = BackgroundListener.class.getSimpleName();

    public BackgroundListener() {

    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    void onBackground() {
        QBChatService.getInstance().destroy();
        Log.d(TAG, "Going Background");
    }
}

File: sample-chat-kotlin/app/src/main/java/com/quickblox/sample/chat/kotlin/managers/BackgroundListener.kt
package com.quickblox.sample.chat.kotlin.managers

import android.util.Log
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import com.quickblox.sample.chat.kotlin.utils.chat.ChatHelper

class BackgroundListener : LifecycleObserver {

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    internal fun onBackground() {
        ChatHelper.destroy()
        Log.d("BackgroundListener", "Background Successful")
    }
}